pep-9999.rst

```diff
@@ -0,0 +1,874 @@
+[#cd]_ project that cover all packages from Pypi, Maven, npm and Rubygems.
+ClearlyDefined is an open source project to help improve clarity of other open
+source projecs that is incubating at the OSI (Open Source Initiative) [#osi]_.
+
+
+Rationale
+=========
+
+A mini survey of existing license metadata definitions in use in Python today
+and documented in several other system/distro and application package formats is
+provided as an informative Appendix in this PEP.
+
+There are a few takeaways from this survey:
+
+- Most package formats use a single `license` field.
+
+- Many modern package formats use some form of license expression syntax to
+  optionally combine more than one license identifiers together. SPDX and
+  SPDX-like syntaxes are the most popular in use.
+
+- SPDX license ids are becoming a de-facto way to reference common licenses
+  everywhere, whether or not a license expression syntax is used. But they often
+  need to be supplemented with extra license ids or conventions to accept extra
+  or generic licenses such as "Proprietary" or "Public domain" not tracked by
+  SPDX.
+
+- Several package format support documenting both a license expression and
+  the paths of the corresponding files that contain the license text. Most free
+  and open source software licenses require to include their full text in a
+  distribution.
+
+These considerations have guided the design and updates proposed in this PEP.
+
+Reusing the existing `license` field with license expressions will provide a an
+intuitive and more structured way to express the license of a distribution using
+a well defined syntax and well known license ids.
+
+Over time, recommending the usage of these expressions will help Python package
+publishers improve the clarity of their license documentation to the benefit of
+packages authors, consumers and redistributors.
+
+
+Core Metadata Specification updates
+===================================
+
+The canonical source for the names and semantics of each of the supported
+metadata fields is the Core Metadata Specification [#cms]_ document.
+
+The details of the updates considered to the Core Metadata Specification [#cms]_
+document as part of his PEP are detailed here and will be added to the
+canonical source once this PEP is approved.
+
+
+Added in Version 2.2
+--------------------
+
+License-File (multiple use)
+:::::::::::::::::::::::::::
+
+The is a string that is a package-root relative path to a license file. The
+license file content __must__ be UTF-8-encoded text.
+
+Build tools should honor this field and include the corresponding license
+file(s) in the built package.
+
+Note that this is field already exists in `wheel` and `setuptools` with this
+behaviour. Therefore this addition to the Core metadata is only recognizing and
+documenting an existing practice as used in `wheels` (with the `license_file`
+and `license_files` `setup.cfg` `[metadata]` entries) and in `setuptools`
```

Comment:
This paragraph doesn't belong in the specification (it's fine to have it somewhere else in the PEP as part of the rationale).

`setup.cfg` also isn't part of the `wheel` format per se - it's solely a `setuptools` input file (the `bdist_wheel` setuptools command is just the original way of making wheel archives).

Reply:
Let me move this out. This is redundant anyway as I already mention this elsewhere.

Reply:
This has moved out in https://github.com/pombredanne/spdx-pypi-pep/pull/2/commits/3200dfbb7945df1a74561c22e58cc1cfdec26ffb
Thank you!